### PR TITLE
perl5db: distinguish an empty list or undef for w expressions

### PR DESCRIPTION
**Pull request by tonycoz**
The debugger before the re-work around ~2012 discarded any list value returned by the expression beyond the first element, but it did distinguish an empty string from undef for that single value.

The re-work attempted to fix that by join()ing the elements returned but this join was done before the filtering to distinguish undef from an empty string, which resulted in watch expressions not stopping on a change from undef to an empty string (or back).

So instead, filter each element returned by the watch expression for undef before doing the join.

### CODE REVIEW

**Comment by jkeenan** [comment]
@tonycoz, is there any simple program you could provide that I could call in the debugger before/after applying this p.r. and that would illustrate the intended differences?  Thanks.

**Comment by tonycoz** [comment]
The code in the test case should do that.

**Comment by khwilliamson** [comment]
Should this be merged now?

**Comment by tonycoz** [comment]
It's just waiting on approval.

**Comment by khwilliamson** [comment]
We cannot let warnocking get in the way of progress.  On my pull requests, if I don't hear any response in a reasonable time, I'm going to push.  If I have some uncertaintly, I would give a deadline in  a GH comment.

**Review by khwilliamson — COMMENTED** [review]
lgtm, but I'm no expert here

**Review by mauke — APPROVED** [review]

**Comment by tonycoz** [comment]
Fixes #22207

**Pull request merged**
